library/ssl_tls13_generic.c

```diff
@@ -1749,6 +1752,36 @@ int mbedtls_ssl_tls13_parse_record_size_limit_ext(mbedtls_ssl_context *ssl,
     return 0;
 }
 
+MBEDTLS_CHECK_RETURN_CRITICAL
+int mbedtls_ssl_tls13_write_record_size_limit_ext(mbedtls_ssl_context *ssl,
+                                                  unsigned char *buf,
+                                                  const unsigned char *end,
+                                                  size_t *out_len)
+{
+    unsigned char *p = buf;
+    *out_len = 0;
+
+    MBEDTLS_STATIC_ASSERT(MBEDTLS_SSL_IN_CONTENT_LEN >= MBEDTLS_SSL_RECORD_SIZE_LIMIT_MIN,
+                          "MBEDTLS_SSL_IN_CONTENT_LEN is less than the "
+                          "minimum record size limit");
```

Comment:
That seems reasonable to me. As you said it is not really possible to put that in check_config.h. It is a build time error like it would be in `check_config.h`. The upper limit is not tested in check_config.h but in ssl_misc.h:
```
if MBEDTLS_SSL_IN_CONTENT_LEN > 16384                                          
#error "Bad configuration - incoming record content too large."                 
#endif
```
When addressing #7012 this should be moved at least to the configuration function. Maybe to ssl_misc.h together with the upper limit check?

Reply:
Yes, `ssl_misc.h` seems the right place, and yes, doing it as part of #7012.

Also: this should probably be a pre-processor check rather than a static assert: I would use a static assert if (for example) there was data in a buffer of size `MBEDTLS_SSL_RECORD_SIZE_LIMIT_MIN` that was going to be copied into a buffer of size `MBEDTLS_SSL_IN_CONTENT_LEN` by this function; but for configuration checks we typically use #if ...